File: Product_APP_MVC/app/src/main/java/com/example/product_app_mvc/DB/Fav_Product_DB_DAO.java
package com.example.product_app_mvc.DB;

import static androidx.room.OnConflictStrategy.IGNORE;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;

import com.example.product_app_mvc.model.POJO_class;

import java.util.List;

/** @noinspection AndroidUnresolvedRoomSqlReference*/
@Dao
public interface Fav_Product_DB_DAO {
    //obsevable
    @Query("SELECT * FROM FavProducts")
    LiveData<List<POJO_class>> getFavProduct();

    @Insert(onConflict = IGNORE)
    void addFavProduct(POJO_class MyProduct);

    @Delete
    void removeFavProduct(POJO_class MyProduct);
}

File: Android_Development/05_Android_Kotlin/Myapp/MyApplication/app/src/main/java/com/example/myapplication/database/Fav_Product_DB_DAO.kt
package com.example.myapplication.database

import androidx.lifecycle.LiveData
import androidx.room.*
import com.example.myapplication.model.Product
import kotlinx.coroutines.flow.Flow

@Dao
interface Fav_Product_DB_DAO {

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    suspend fun insertProduct(product: Product)

    @Delete
    suspend fun deleteProduct(product: Product)

    @Query("SELECT * FROM product_table")
    fun getAllFavProducts(): Flow<List<Product>>

    @Query("SELECT * FROM product_table WHERE id = :id LIMIT 1")
    suspend fun getProductById(id: Int): Product?
}
